aptos-move/framework/aptos-framework/sources/account_recovery.move

```diff
@@ -0,0 +1,148 @@
+    use std::vector;
+    use std::option::Option;
+    use std::option;
+    use aptos_framework::resource_account;
+    use aptos_framework::account::SignerCapability;
+
+    /// Account recovery struct
+    struct AccountRecovery has key, store, drop {
+        authorized_addresses: vector<address>,
+        required_delay_seconds: u64,
+        account_recovery_init: Option<AccountRecoveryInitData>,
+        // TODO add recovery capability here.
+    }
+
+    // Reverse lookup map
+    struct AccountRecoveryReverseLookup has key, store {
+        authorized_to_recovery: Table<address, vector<address>>,
+    }
+
+    struct AccountRecoveryInitData has key, store {
+        recovery_seq_number: u64,
+        recovery_initiation_ts: u64,
+    }
+
+    struct ModuleData has key {
+        // Storing the signer capability here, so the module can programmatically sign for transactions
+        signer_cap: SignerCapability,
+    }
+
+
+    fun initialize(signer: &signer) acquires AccountRecoveryReverseLookup {
+        move_to(signer, AccountRecoveryReverseLookup {
+            authorized_to_recovery: table::new(),
+        });
+
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(signer, @source_addr);
+
+        // Store the token data id and the resource account's signer capability within the module, so we can programmatically
+        // sign for transactions in the `mint_event_ticket()` function.
+        move_to(signer, ModuleData {
+            signer_cap: resource_signer_cap,
+        });
+    }
+
+    fun recovery_exists(addr: address): bool {
+        exists<AccountRecovery>(addr)
+    }
+
+    public fun register(account: &signer,
+                        authorized_addresses: vector<address>,
+                        required_delay_seconds: u64,
+                        rotation_capability_sig_bytes: vector<u8>,
+                        account_public_key_bytes: vector<u8>,
+    ) acquires AccountRecovery, AccountRecoveryReverseLookup {
+        let addr = signer::address_of(account);
+
+        assert!(!recovery_exists(addr), ERECOVERY_ALREADY_SET);
+        assert!(!exists<AccountRecoveryReverseLookup>(@source_addr));
+
+        let reverse_lookup = borrow_global_mut<AccountRecoveryReverseLookup>(@source_addr);
+
+        move_to(account, AccountRecovery {
+            authorized_addresses,
+            required_delay_seconds,
+            account_recovery_init: option::none<>(),
+        });
+
+        let i = 0;
+        let len = vector::length(&authorized_addresses);
+        while (i < len) {
+            let authorized_address = *vector::borrow(&authorized_addresses, i);
+            let list = table::borrow_mut_with_default(&mut reverse_lookup.authorized_to_recovery, authorized_address, vector::empty<address>());
+            vector::push_back(list, addr);
+        };
+        account::offer_rotation_capability(signer, rotation_capability_sig_bytes, 0, account_public_key_bytes, @source_addr)
```

Comment:
we need to use the resource account's address for this - also for all of the
```
!exists<AccountRecoveryReverseLookup>(@source_addr)
```
calls above, it should be the resource account's address (in this case `@aptos_framework` but I don't think there's a way to publish this under `aptos_framework`. maybe move it to somewhere else and give it another address name? )

Reply:
nope, `@source_addr` is a named_address - usually you need a `Move.toml` file and put the `@source_addr` as a name address there, otherwise it wouldn't compile. the name/address of the module is `@aptos_framework` in this case - if you move this module to something like `hackathon::account_recovery`, the name/address of the module is `@hackathon`.